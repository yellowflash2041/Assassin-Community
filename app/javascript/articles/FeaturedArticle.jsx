import { h } from 'preact';
import PropTypes from 'prop-types';
import {
  TagList,
  SaveButton,
  ReadingTime,
  ReactionsCount,
  CommentsCount,
} from './components';
import { articlePropTypes } from '../src/components/common-prop-types';

export const FeaturedArticle = ({
  article,
  isBookmarked,
  reactionsIcon,
  commentsIcon,
  bookmarkClick,
}) => {
  return (
    <div>
      <div
        id="featured-story-marker"
        data-featured-article={`articles-${article.id}`}
      />
      <img
        src={article.main_image}
        style={{ display: 'none' }}
        alt={article.title}
      />
      <a
        href={article.path}
        id={`article-link-${article.id}`}
        className="index-article-link"
        aria-label="Main Story"
        data-featured-article={`articles-${article.id}`}
      />
      <div
        className="single-article big-article"
        data-content-user-id={article.user_id}
      >
        <a
          href={article.path}
          id={`article-link-${article.id}`}
          className="index-article-link"
          aria-label="Main Story"
          data-featured-article={`articles-${article.id}`}
        >
          <div
            className="picture image-final"
            style={{
              backgroundColor: article.main_image_background_hex_color,
              backgroundImage: `url(${article.main_image})`,
            }}
          />
          <div className="content-wrapper">
            <h3>{article.title}</h3>
          </div>
        </a>
        <a
          href={`/${article.user.username}`}
          className="featured-profile-button"
        >
          <img
            className="featured-profile-pic"
            src={article.user.profile_image_90}
            alt={article.title}
          />
        </a>
        <div className="featured-user-name">
          <a href={`/${article.user.username}`}>
            {article.user.name} ãƒ»{' '}
            <time dateTime={article.published_timestamp}>
              {article.readable_publish_date}
            </time>
            <span
              className="time-ago-indicator-initial-placeholder"
              data-seconds={`${article.published_at_int}`}
            />
          </a>
        </div>
        <TagList tags={article.tag_list} className="featured-tags" />
        <CommentsCount
          count={article.comments_count}
          articlePath={article.path}
          icon={commentsIcon}
          className="featured-engagement-count"
        />
        <ReactionsCount article={article} icon={reactionsIcon} />
        <ReadingTime
          articlePath={article.path}
          readingTime={article.reading_time}
        />
        <SaveButton
          article={article}
          isBookmarked={isBookmarked}
          onClick={bookmarkClick}
        />
      </div>
    </div>
  );
};

FeaturedArticle.defaultProps = {
  isBookmarked: false,
};

FeaturedArticle.propTypes = {
  article: articlePropTypes.isRequired,
  isBookmarked: PropTypes.bool,
  reactionsIcon: PropTypes.string.isRequired,
  commentsIcon: PropTypes.string.isRequired,
  bookmarkClick: PropTypes.func.isRequired,
};
