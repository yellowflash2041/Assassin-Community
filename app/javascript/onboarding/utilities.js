export const jsonToForm = data => {
  const form = new FormData();
  data.forEach(item => form.append(item.key, item.value));
  return form;
};

export const getContentOfToken = token =>
  document.querySelector(`meta[name='${token}']`).content;
