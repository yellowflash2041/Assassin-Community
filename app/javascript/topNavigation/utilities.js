function closeHeaderMenu(memberMenu, menuNavButton) {
  if (menuNavButton) {
    menuNavButton.setAttribute('aria-expanded', 'false');
  }

  if (memberMenu) {
    setTimeout(() => {
      memberMenu.classList.remove('showing');
    }, 5);
  }
}

function blurHeaderMenu(memberMenu, menuNavButton, potentiallyActiveElement) {
  setTimeout(() => {
    if (document.activeElement !== potentiallyActiveElement) {
      closeHeaderMenu(memberMenu, menuNavButton);
    }
  }, 10);
}

function toggleHeaderMenu(memberMenu, navigationButton) {
  if (!memberMenu || !navigationButton) {
    return;
  }

  let crayonsHeaderMenuClassList = memberMenu.classList;
  if (crayonsHeaderMenuClassList.contains('showing')) {
    crayonsHeaderMenuClassList.remove('showing');
    navigationButton.setAttribute('aria-expanded', 'false');
  } else {
    crayonsHeaderMenuClassList.add('showing');
    navigationButton.setAttribute('aria-expanded', 'true');

    let firstNavLink = document.getElementById('first-nav-link');
    if (firstNavLink) {
      setTimeout(() => {
        // focus first item on open
        firstNavLink.focus();
      }, 100);
    }
  }
}

export function isTouchDevice() {
  return /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini|DEV-Native-ios/i.test(
    navigator.userAgent,
  );
}

export function initializeTouchDevice(memberTopMenu, menuNavButton) {
  if (navigator.userAgent === 'DEV-Native-ios') {
    document.body.classList.add('dev-ios-native-body');
  }

  if (memberTopMenu) {
    let crayonsHeaderMenuClassList = memberTopMenu.classList;

    setTimeout(() => {
      closeHeaderMenu(memberTopMenu, menuNavButton);

      if (isTouchDevice()) {
        // Use a named function instead of anonymous so duplicate event handlers are discarded
        menuNavButton.addEventListener('click', (_event) => {
          toggleHeaderMenu(memberTopMenu, menuNavButton);
        });
      } else {
        crayonsHeaderMenuClassList.add('desktop');
        menuNavButton.addEventListener('click', (_event) => {
          toggleHeaderMenu(memberTopMenu, menuNavButton);
        });
        memberTopMenu.addEventListener('keyup', (e) => {
          if (
            e.key === 'Escape' &&
            crayonsHeaderMenuClassList.contains('showing')
          ) {
            crayonsHeaderMenuClassList.remove('showing');
            menuNavButton.focus();
          }
        });
        document
          .getElementById('last-nav-link')
          .addEventListener('blur', (_event) => {
            blurHeaderMenu(
              memberTopMenu,
              menuNavButton,
              document.getElementById('second-last-nav-link'),
            );
          });
        document.addEventListener('click', (_event) => {
          // if clicking outside of the menu, close it
          if (!memberTopMenu.contains(document.activeElement)) {
            blurHeaderMenu(
              memberTopMenu,
              menuNavButton,
              document.getElementById('first-nav-link'),
            );
          }
        });
      }
    }, 10);
  }
}

function toggleBurgerMenu() {
  document.body.classList.toggle('hamburger-open');
}

function showMoreMenu({ target }) {
  target.nextElementSibling.classList.remove('hidden');
  target.classList.add('hidden');
}

/**
 * Gets a reference to InstantClick
 *
 * @param {number} [waitTime=2000] The amount of time to wait
 * until giving up waiting for InstantClick to exist
 *
 * @returns {Promise<object>} The global instance of InstantClick.
 */
export async function getInstantClick(waitTime = 2000) {
  return new Promise((resolve, reject) => {
    const failTimer = setTimeout(() => {
      clearInterval(timer);
      reject(new Error('Unable to resolve InstantClick'));
    }, waitTime);

    const timer = setInterval(() => {
      if (typeof InstantClick !== 'undefined') {
        clearTimeout(failTimer);
        clearInterval(timer);
        resolve(InstantClick);
      }
    });
  });
}

/**
 * Initializes the hamburger menu for mobile navigation
 *
 * @param {HTMLElement[]} menus
 * @param {HTMLElement[]} moreMenus
 */
export function initializeMobileMenu(menus, moreMenus) {
  menus.forEach((trigger) => {
    trigger.addEventListener('click', toggleBurgerMenu);
  });

  moreMenus.forEach((trigger) => {
    trigger.addEventListener('click', showMoreMenu);
  });
}

/**
 * Sets the icon link visually for the current page if the current page
 * is one of the main icon links of the top navigation.
 *
 * @param {string} currentPage
 * @param {[string, HTMLElement][]} pageEntries
 */
export function setCurrentPageIconLink(currentPage, pageEntries) {
  pageEntries
    // Filter out nulls (means the user is logged out so most icons are not in the logged out view)
    .filter(([, iconLink]) => iconLink)
    .forEach(([page, iconLink]) => {
      if (currentPage === page) {
        iconLink.blur();
        iconLink.classList.add('crayons-header__link--current');
      } else {
        iconLink.classList.remove('crayons-header__link--current');
      }
    });
}
