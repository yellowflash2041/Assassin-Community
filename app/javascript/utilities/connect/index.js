export * from './getUnopenedChannels';
export * from './pusher';
