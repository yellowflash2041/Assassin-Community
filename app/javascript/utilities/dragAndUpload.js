import { generateMainImage } from '../article-form/actions';
import { validateFileInputs } from '../packs/validateFileInputs';

export function dragDrop(files, handleImageSuccess, handleImageFailure) {
  if (files.length > 0 && validateFileInputs()) {
    const payload = { image: files };
    generateMainImage(payload, handleImageSuccess, handleImageFailure);
  }
}
