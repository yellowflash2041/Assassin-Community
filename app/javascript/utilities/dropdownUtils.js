/**
 * Helper query string to identify interactive/focusable HTML elements
 */
const INTERACTIVE_ELEMENTS_QUERY =
  'button, [href], input:not([type="hidden"]), select, textarea, [tabindex="0"]';

/**
 * Used to close the given dropdown if:
 * - Escape is pressed
 * - Tab is pressed and the newly focused element doesn't exist inside the dropdown
 *
 * @param {Object} args
 * @param {string} args.triggerElementId The id of the button which activates the dropdown
 * @param {string} args.dropdownContentId The id of the dropdown content element
 * @param {Function} args.onClose Optional function for any side-effects which should occur on dropdown close
 */
const keyUpListener = ({ triggerElementId, dropdownContentId, onClose }) => {
  return ({ key }) => {
    if (key === 'Escape') {
      // Close the dropdown and return focus to the trigger button to prevent focus being lost
      const triggerElement = document.getElementById(triggerElementId);
      const isCurrentlyOpen =
        triggerElement.getAttribute('aria-expanded') === 'true';
      if (isCurrentlyOpen) {
        closeDropdown({ triggerElementId, dropdownContentId, onClose });
        triggerElement.focus();
      }
    } else if (key === 'Tab') {
      // Close the dropdown if the user has tabbed away from it
      const isInsideDropdown = document
        .getElementById(dropdownContentId)
        ?.contains(document.activeElement);
      if (!isInsideDropdown) {
        closeDropdown({ triggerElementId, dropdownContentId, onClose });
      }
    }
  };
};

/**
 * Used to listen for a click outside of a dropdown while it's open.
 * Closes the dropdown and refocuses the trigger button, if another interactive item has not been clicked.
 *
 * @param {Object} args
 * @param {string} args.triggerElementId The id of the button which activates the dropdown
 * @param {string} args.dropdownContentId The id of the dropdown content element
 * @param {Function} args.onClose Optional function for any side-effects which should occur on dropdown close
 */
const clickOutsideListener = ({
  triggerElementId,
  dropdownContentId,
  onClose,
}) => {
  return ({ target }) => {
    const triggerElement = document.getElementById(triggerElementId);
    const dropdownContent = document.getElementById(dropdownContentId);
    if (!dropdownContent) {
      // User may have navigated away from the page
      return;
    }

    if (
      target !== triggerElement &&
      !dropdownContent.contains(target) &&
      !triggerElement.contains(target)
    ) {
      closeDropdown({ triggerElementId, dropdownContentId, onClose });

      //   If the user did not click on another interactive item, return focus to the trigger
      if (!target.matches(INTERACTIVE_ELEMENTS_QUERY)) {
        triggerElement.focus();
      }
    }
  };
};

/**
 * Open the given dropdown, updating aria attributes, attaching listeners and focus the first interactive element
 *
 * @param {Object} args
 * @param {string} args.triggerElementId The id of the button which activates the dropdown
 * @param {string} args.dropdownContent The id of the dropdown content element
 * @param {Function} args.onClose Optional function for any side-effects which should occur on dropdown close
 */
const openDropdown = ({ triggerElementId, dropdownContentId, onClose }) => {
  const dropdownContent = document.getElementById(dropdownContentId);
  const triggerElement = document.getElementById(triggerElementId);

  triggerElement.setAttribute('aria-expanded', 'true');

  // Style set inline to prevent specificity issues
  dropdownContent.style.display = 'block';

  // Send focus to the first suitable element
  dropdownContent.querySelector(INTERACTIVE_ELEMENTS_QUERY)?.focus();

  document.addEventListener(
    'keyup',
    keyUpListener({ triggerElementId, dropdownContentId, onClose }),
  );

  document.addEventListener(
    'click',
    clickOutsideListener({ triggerElementId, dropdownContentId, onClose }),
  );
};

/**
 * Close the given dropdown, updating aria attributes and removing event listeners
 *
 * @param {Object} args
 * @param {string} args.triggerElementId The id of the button which activates the dropdown
 * @param {string} args.dropdownContent The id of the dropdown content element
 * @param {Function} args.onClose Optional function for any side-effects which should occur on dropdown close
 */
const closeDropdown = ({ triggerElementId, dropdownContentId, onClose }) => {
  const dropdownContent = document.getElementById(dropdownContentId);

  document
    .getElementById(triggerElementId)
    ?.setAttribute('aria-expanded', 'false');

  // Remove the inline style added when we opened the dropdown
  dropdownContent.style.removeProperty('display');

  document.removeEventListener(
    'keyup',
    keyUpListener({ triggerElementId, dropdownContentId, onClose }),
  );

  document.removeEventListener(
    'click',
    clickOutsideListener({ triggerElementId, dropdownContentId, onClose }),
  );
  onClose?.();
};

/**
 * A helper function to initialize dropdown behaviors. This function attaches open/close click and keyup listeners,
 * and makes sure relevant aria properties and keyboard focus are updated.
 *
 * @param {Object} args
 * @param {string} args.triggerButtonElementId The ID of the button which triggers the dropdown open/close behavior
 * @param {string} args.dropdownContentId The ID of the dropdown content which should open/close on trigger button press
 * @param {string} args.dropdownContentCloseButtonId Optional ID of any button within the dropdown content which should close the dropdown
 * @param {Function} args.onClose An optional callback for when the dropdown is closed. This can be passed to execute any side-effects required when the dropdown closes.
 * @param {Function} args.onOpen An optional callback for when the dropdown is opened. This can be passed to execute any side-effects required when the dropdown opens.
 *
 * @returns {{closeDropdown: Function}} Object with callback to close the initialized dropdown
 */
export const initializeDropdown = ({
  triggerElementId,
  dropdownContentId,
  dropdownContentCloseButtonId,
  onClose,
  onOpen,
}) => {
  const triggerButton = document.getElementById(triggerElementId);
  const dropdownContent = document.getElementById(dropdownContentId);

  if (!triggerButton || !dropdownContent) {
    // The required props haven't been provided, do nothing
    return;
  }

  // Ensure default values have been applied
  triggerButton.setAttribute('aria-expanded', 'false');
  triggerButton.setAttribute('aria-controls', dropdownContentId);
  triggerButton.setAttribute('aria-haspopup', 'true');

  triggerButton.addEventListener('click', () => {
    if (
      document
        .getElementById(triggerElementId)
        ?.getAttribute('aria-expanded') === 'true'
    ) {
      closeDropdown({
        triggerElementId,
        dropdownContentId,
        onClose,
      });
    } else {
      openDropdown({
        triggerElementId,
        dropdownContentId,
        onClose,
      });
      onOpen?.();
    }
  });

  if (dropdownContentCloseButtonId) {
    // The dropdown content has a 'close' button inside that we also need to handle
    document
      .getElementById(dropdownContentCloseButtonId)
      ?.addEventListener('click', () => {
        closeDropdown({
          triggerElementId,
          dropdownContentId,
          onClose,
        });
        document.getElementById(triggerElementId)?.focus();
      });
  }

  return {
    closeDropdown: () =>
      closeDropdown({
        triggerElementId,
        dropdownContentId,
        onClose,
      }),
  };
};
