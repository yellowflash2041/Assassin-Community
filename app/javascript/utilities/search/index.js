// TODO: We should really be using the xss package by installing it in package.json
// but for now filterXSS is global because of legacy JS

import { request } from '../http';

function getParameterByName(name, url = window.location.href) {
  const sanitizedName = name.replace(/[[\]]/g, '\\$&');
  const regex = new RegExp(`[?&]${sanitizedName}(=([^&#]*)|&|#|$)`);
  const results = regex.exec(url);

  if (!results) {
    return null;
  }

  if (!results[2]) {
    return '';
  }

  return decodeURIComponent(results[2].replace(/\+/g, ' '));
}

function getParameters(name, url) {
  const params = getParameterByName(name, url);

  if (params) {
    return `&${name}=${params}`;
  }

  return '';
}

function getFilterParameters(url) {
  return getParameters('filters', url);
}

function getSortParameters(url) {
  const sortBy = getParameters('sort_by', url);
  const sortDirection = getParameters('sort_direction', url);

  return sortBy + sortDirection;
}

export const hasInstantClick = () => typeof instantClick !== 'undefined';

function fixedEncodeURIComponent(str) {
  // from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent
  return encodeURIComponent(str).replace(
    /[!'()*]/g,
    (c) => `%${c.charCodeAt(0).toString(16)}`,
  );
}

export function displaySearchResults({
  searchTerm,
  location = window.location,
}) {
  const baseUrl = location.origin;
  const sanitizedQuery = fixedEncodeURIComponent(searchTerm);
  const filterParameters = getFilterParameters(location.href);
  const sortParameters = getSortParameters(location.href);

  InstantClick.display(
    `${baseUrl}/search?q=${sanitizedQuery}${filterParameters}${sortParameters}`,
  );
}

export function getInitialSearchTerm(querystring) {
  const matches = /(?:&|\?)?q=([^&=]+)/.exec(querystring);
  const rawSearchTerm =
    matches !== null && matches.length === 2
      ? decodeURIComponent(matches[1].replace(/\+/g, '%20'))
      : '';
  const query = filterXSS(rawSearchTerm);
  const divForDecode = document.createElement('div');
  divForDecode.innerHTML = query;

  return divForDecode.firstChild !== null
    ? divForDecode.firstChild.nodeValue
    : query;
}

export function preloadSearchResults({
  searchTerm,
  location = window.location,
}) {
  const encodedQuery = fixedEncodeURIComponent(
    searchTerm.replace(/^[ ]+|[ ]+$/g, ''),
  );
  InstantClick.preload(
    `${location.origin}/search?q=${encodedQuery}${getFilterParameters(
      location.href,
    )}`,
  );
}

export function createSearchUrl(dataHash) {
  const searchParams = new URLSearchParams();
  Object.keys(dataHash).forEach((key) => {
    const value = dataHash[key];
    if (Array.isArray(value)) {
      value.forEach((arrayValue) => {
        searchParams.append(`${key}[]`, arrayValue);
      });
    } else {
      searchParams.append(key, value);
    }
  });

  return searchParams.toString();
}

/**
 * A helper method to call /search endpoints.
 *
 * @param {string} endpoint - The search endpoint you want to request (i.e. tags).
 * @param {object} dataHash - A hash with the search params that need to be included in the request.
 *
 * @returns {Promise} A promise object with response formatted as JSON.
 */
export function fetchSearch(endpoint, dataHash) {
  const searchUrl = createSearchUrl(dataHash);

  return request(`/search/${endpoint}?${searchUrl}`).then((response) =>
    response.json(),
  );
}
