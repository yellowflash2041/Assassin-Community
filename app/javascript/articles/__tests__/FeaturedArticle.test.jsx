import { h } from 'preact';
import render from 'preact-render-to-json';
import { featuredArticle, assetPath } from './utilities/articleUtilities';
import '../../../assets/javascripts/lib/xss';
import '../../../assets/javascripts/utilities/timeAgo';
import { FeaturedArticle } from '../FeaturedArticle';

const featuredArticleNoReactions = {
  ...featuredArticle,
  positive_reactions_count: 0,
};
const commonProps = {
  reactionsIcon: assetPath('reactions-stack.png'),
  commentsIcon: assetPath('comments-bubble.png'),
};

describe('<FeaturedArticle /> component', () => {
  it('should render a standard article', () => {
    const tree = render(
      <FeaturedArticle
        {...commonProps}
        isBookmarked={false}
        article={featuredArticle}
        currentTag="javascript"
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render with the reactions icon when the reactions count is 0', () => {
    const tree = render(
      <FeaturedArticle
        {...commonProps}
        isBookmarked={false}
        article={featuredArticleNoReactions}
        currentTag="javascript"
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render a bookmarked featured article', () => {
    const tree = render(
      <FeaturedArticle
        {...commonProps}
        isBookmarked
        article={featuredArticle}
        currentTag="javascript"
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
