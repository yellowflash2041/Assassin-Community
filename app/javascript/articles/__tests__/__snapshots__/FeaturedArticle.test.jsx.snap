// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedArticle /> component should render a bookmarked featured article 1`] = `
<div>
  <div
    data-featured-article="articles-62407"
    id="featured-story-marker"
  />
  <img
    alt="Unbranded Home Loan Account"
    src="/images/onboarding-background.png"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <a
    aria-label="Main Story"
    class="index-article-link"
    data-featured-article="articles-62407"
    href="/some-post/path"
    id="article-link-62407"
  />
  <div
    class="single-article big-article"
    data-content-user-id={23289}
  >
    <a
      aria-label="Main Story"
      class="index-article-link"
      data-featured-article="articles-62407"
      href="/some-post/path"
      id="article-link-62407"
    >
      <div
        class="picture image-final"
        style={
          Object {
            "backgroundColor": undefined,
            "backgroundImage": "url(/images/onboarding-background.png)",
          }
        }
      />
      <div
        class="content-wrapper"
      >
        <h3>
          Unbranded Home Loan Account
        </h3>
      </div>
    </a>
    <a
      class="featured-profile-button"
      href="/Emil99"
    >
      <img
        alt="Unbranded Home Loan Account"
        class="featured-profile-pic"
        src="/images/10.png"
      />
    </a>
    <div
      class="featured-user-name"
    >
      <a
        href="/Emil99"
      >
        Stella Macejkovic ・ 
        <time
          dateTime="Tue, 18 Feb 2020 14:59:24 GMT"
        >
          February 18
        </time>
        <span
          class="time-ago-indicator-initial-placeholder"
          data-seconds="1582037964819"
        />
      </a>
    </div>
    <div
      class="tags featured-tags"
    >
      <a
        href="/t/javascript"
      >
        <span
          class="tag"
        >
          #javascript
        </span>
      </a>
      <a
        href="/t/ruby"
      >
        <span
          class="tag"
        >
          #ruby
        </span>
      </a>
      <a
        href="/t/go"
      >
        <span
          class="tag"
        >
          #go
        </span>
      </a>
    </div>
    <div
      class="article-engagement-count comments-count featured-engagement-count"
    >
      <a
        href="/some-post/path#comments"
      >
        <img
          alt="chat"
          loading="lazy"
          src="/images/comments-bubble.png"
        />
        <span
          class="engagement-count-number"
        >
          213
        </span>
      </a>
    </div>
    <div
      class="article-engagement-count reactions-count"
    >
      <a
        href="/some-post/path"
      >
        <img
          alt="heart"
          loading="lazy"
          src="/images/reactions-stack.png"
        />
        <span
          class="engagement-count-number"
          id="engagement-count-number-62407"
        >
          428
        </span>
      </a>
    </div>
    <a
      class="article-reading-time"
      href="/some-post/path"
    >
      undefined min read
    </a>
    <button
      class="article-engagement-count engage-button bookmark-button selected"
      data-initial-feed={true}
      data-reactable-id={62407}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOver={[Function]}
      onMouseout={[Function]}
      type="button"
    >
      <span
        class="bm-success"
      />
    </button>
  </div>
</div>
`;

exports[`<FeaturedArticle /> component should render a standard article 1`] = `
<div>
  <div
    data-featured-article="articles-62407"
    id="featured-story-marker"
  />
  <img
    alt="Unbranded Home Loan Account"
    src="/images/onboarding-background.png"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <a
    aria-label="Main Story"
    class="index-article-link"
    data-featured-article="articles-62407"
    href="/some-post/path"
    id="article-link-62407"
  />
  <div
    class="single-article big-article"
    data-content-user-id={23289}
  >
    <a
      aria-label="Main Story"
      class="index-article-link"
      data-featured-article="articles-62407"
      href="/some-post/path"
      id="article-link-62407"
    >
      <div
        class="picture image-final"
        style={
          Object {
            "backgroundColor": undefined,
            "backgroundImage": "url(/images/onboarding-background.png)",
          }
        }
      />
      <div
        class="content-wrapper"
      >
        <h3>
          Unbranded Home Loan Account
        </h3>
      </div>
    </a>
    <a
      class="featured-profile-button"
      href="/Emil99"
    >
      <img
        alt="Unbranded Home Loan Account"
        class="featured-profile-pic"
        src="/images/10.png"
      />
    </a>
    <div
      class="featured-user-name"
    >
      <a
        href="/Emil99"
      >
        Stella Macejkovic ・ 
        <time
          dateTime="Tue, 18 Feb 2020 14:59:24 GMT"
        >
          February 18
        </time>
        <span
          class="time-ago-indicator-initial-placeholder"
          data-seconds="1582037964819"
        />
      </a>
    </div>
    <div
      class="tags featured-tags"
    >
      <a
        href="/t/javascript"
      >
        <span
          class="tag"
        >
          #javascript
        </span>
      </a>
      <a
        href="/t/ruby"
      >
        <span
          class="tag"
        >
          #ruby
        </span>
      </a>
      <a
        href="/t/go"
      >
        <span
          class="tag"
        >
          #go
        </span>
      </a>
    </div>
    <div
      class="article-engagement-count comments-count featured-engagement-count"
    >
      <a
        href="/some-post/path#comments"
      >
        <img
          alt="chat"
          loading="lazy"
          src="/images/comments-bubble.png"
        />
        <span
          class="engagement-count-number"
        >
          213
        </span>
      </a>
    </div>
    <div
      class="article-engagement-count reactions-count"
    >
      <a
        href="/some-post/path"
      >
        <img
          alt="heart"
          loading="lazy"
          src="/images/reactions-stack.png"
        />
        <span
          class="engagement-count-number"
          id="engagement-count-number-62407"
        >
          428
        </span>
      </a>
    </div>
    <a
      class="article-reading-time"
      href="/some-post/path"
    >
      undefined min read
    </a>
    <button
      class="article-engagement-count engage-button bookmark-button "
      data-initial-feed={true}
      data-reactable-id={62407}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOver={[Function]}
      onMouseout={[Function]}
      type="button"
    >
      <span
        class="bm-initial"
      />
    </button>
  </div>
</div>
`;

exports[`<FeaturedArticle /> component should render with the reactions icon when the reactions count is 0 1`] = `
<div>
  <div
    data-featured-article="articles-62407"
    id="featured-story-marker"
  />
  <img
    alt="Unbranded Home Loan Account"
    src="/images/onboarding-background.png"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <a
    aria-label="Main Story"
    class="index-article-link"
    data-featured-article="articles-62407"
    href="/some-post/path"
    id="article-link-62407"
  />
  <div
    class="single-article big-article"
    data-content-user-id={23289}
  >
    <a
      aria-label="Main Story"
      class="index-article-link"
      data-featured-article="articles-62407"
      href="/some-post/path"
      id="article-link-62407"
    >
      <div
        class="picture image-final"
        style={
          Object {
            "backgroundColor": undefined,
            "backgroundImage": "url(/images/onboarding-background.png)",
          }
        }
      />
      <div
        class="content-wrapper"
      >
        <h3>
          Unbranded Home Loan Account
        </h3>
      </div>
    </a>
    <a
      class="featured-profile-button"
      href="/Emil99"
    >
      <img
        alt="Unbranded Home Loan Account"
        class="featured-profile-pic"
        src="/images/10.png"
      />
    </a>
    <div
      class="featured-user-name"
    >
      <a
        href="/Emil99"
      >
        Stella Macejkovic ・ 
        <time
          dateTime="Tue, 18 Feb 2020 14:59:24 GMT"
        >
          February 18
        </time>
        <span
          class="time-ago-indicator-initial-placeholder"
          data-seconds="1582037964819"
        />
      </a>
    </div>
    <div
      class="tags featured-tags"
    >
      <a
        href="/t/javascript"
      >
        <span
          class="tag"
        >
          #javascript
        </span>
      </a>
      <a
        href="/t/ruby"
      >
        <span
          class="tag"
        >
          #ruby
        </span>
      </a>
      <a
        href="/t/go"
      >
        <span
          class="tag"
        >
          #go
        </span>
      </a>
    </div>
    <div
      class="article-engagement-count comments-count featured-engagement-count"
    >
      <a
        href="/some-post/path#comments"
      >
        <img
          alt="chat"
          loading="lazy"
          src="/images/comments-bubble.png"
        />
        <span
          class="engagement-count-number"
        >
          213
        </span>
      </a>
    </div>
    <div
      class="article-engagement-count reactions-count"
    >
      <a
        href="/some-post/path"
      >
        <img
          alt="heart"
          loading="lazy"
          src="/images/reactions-stack.png"
        />
        <span
          class="engagement-count-number"
          id="engagement-count-number-62407"
        >
          0
        </span>
      </a>
    </div>
    <a
      class="article-reading-time"
      href="/some-post/path"
    >
      undefined min read
    </a>
    <button
      class="article-engagement-count engage-button bookmark-button "
      data-initial-feed={true}
      data-reactable-id={62407}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOver={[Function]}
      onMouseout={[Function]}
      type="button"
    >
      <span
        class="bm-initial"
      />
    </button>
  </div>
</div>
`;
